src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -15,9 +15,11 @@ public class AuthenticationServiceImpl implements AuthenticationService {
     public Driver login(String username, String password)
             throws AuthenticationException {
         Optional<Driver> driver = driverService.findByUsername(username);
-        if (driver.isPresent() && driver.get().getPassword().equals(password)) {
-            return driver.get();
+        Driver result = driver.orElseGet(() -> driver.get() != null
+                && driver.get().getPassword().equals(password) ? driver.get() : null);
+        if (result == null) {
+            throw new AuthenticationException("Username or password was incorrect");
         }
-        throw new AuthenticationException("Username or password was incorrect");
+        return result;
```

Comment:
you have `if` inside `if` basically, try this:
```java
if (!driver.isPresent() || !driver.get().getPassword().equals(password)) {
    throw new AuthenticationException("Username or password was incorrect");
}
return driver.get();
```
or just return it back to what it was before my comment, it is not a big deal anyways :)

Reply:
ok